snapshot-labs/sx-monorepo
# feat: Lido staking

### Summary



Closes: https://github.com/snapshot-labs/pitches/issues/45

![Screenshot 2024-05-11 at 02 44 57](https://github.com/snapshot-labs/sx-monorepo/assets/495709/56b58834-31b5-4b16-80eb-35caf6f641aa)

![Screenshot 2024-05-11 at 02 45 35](https://github.com/snapshot-labs/sx-monorepo/assets/495709/9858db1c-d28e-491e-8176-2871aa44b605)
![Screenshot 2024-05-11 at 02 46 05](https://github.com/snapshot-labs/sx-monorepo/assets/495709/08c97f18-4601-4836-a91f-4c42700517fc)



### How to test

Test space: sep:0x44C51B0f96029b5e03dBABe83849e26A1deCB66d 

Example of executed proposal: https://bafybeiascrg3qrtipklsubtwtbdqk5zwrllbyp6vczrhp6y4bhpwvo5z6y.on.fleek.co/#/sep:0x44C51B0f96029b5e03dBABe83849e26A1deCB66d/proposal/9

1. Go to a space with treasury
2. It should always show the ETH token for eth and sepolia network, even without 0 balance
3. There should be a new STAKE icon next to the ETH token
4. Clicking on the STAKE button should open a modal, prompting for the amount of ETH to stake
5. Clicking on the "max" button will fill the max balance of the ETH token, available in the treasury
6. Clicking on "confirn" should redirect to the create proposal page, with the execution for Lido stake filled
7. You can edit the execution
8. The execution simulation should pass as long as the amount of staked token is valid (<= ETH balance)
9. Create a proposal, and vote for it to pass
10. It should execute the tx, and stake the given amount of ETH on Lido

### Note

- I am not familiar with execution strategies, and may have missed a few scenarios
- The "Stake with lido" strategy is missing from the execution's list in the proposal editor, and this is intentional.
- I have not tested the execution on mainnet, only sepolia
